11/debian-9/rootfs/libpostgresql.sh

```diff
@@ -0,0 +1,699 @@
+    # shellcheck disable=SC2155
+    export NSS_WRAPPER_GROUP="$(mktemp)"
+    echo "postgres:x:$(id -u):$(id -g):PostgreSQL:$POSTGRESQL_DATADIR:/bin/false" > "$NSS_WRAPPER_PASSWD"
+    echo "postgres:x:$(id -g):" > "$NSS_WRAPPER_GROUP"
+  fi
+}
+
+########################
+# Load global variables used on PostgreSQL configuration.
+# Globals:
+#   POSTGRESQL_*
+# Arguments:
+#   None
+# Returns:
+#   Series of exports to be used as 'eval' arguments
+#########################
+postgresql_env() {
+    declare_env_alias() {
+      if env | grep -q "$2"; then
+          cat << EOF 
+export $1="${!2}"
+EOF
+      fi
+    }
+  # Alias created for official postgre image compatibility
+  declare_env_alias POSTGRESQL_PASSWORD POSTGRES_PASSWORD
+  declare_env_alias POSTGRESQL_DATABASE POSTGRES_DB
+  declare_env_alias POSTGRESQL_USERNAME POSTGRES_USER
+  declare_env_alias POSTGRESQL_DATA_DIR PGDATA
+  declare_env_alias POSTGRESQL_INITDB_WALDIR POSTGRES_INITDB_WALDIR
+  declare_env_alias POSTGRESQL_INITDB_ARGS POSTGRES_INITDB_ARGS
+
+  # Alias created for maintain consistency using prefix
+  declare_env_alias POSTGRESQL_CLUSTER_APP_NAME POSTGRES_CLUSTER_APP_NAME
+  declare_env_alias POSTGRESQL_MASTER_HOST POSTGRES_MASTER_HOST
+  declare_env_alias POSTGRESQL_MASTER_PORT_NUMBER POSTGRES_MASTER_PORT_NUMBER
+  declare_env_alias POSTGRESQL_NUM_SYNCHRONOUS_REPLICAS POSTGRES_NUM_SYNCHRONOUS_REPLICAS
+  declare_env_alias POSTGRESQL_PORT_NUMBER POSTGRES_PORT_NUMBER
+  declare_env_alias POSTGRESQL_REPLICATION_MODE POSTGRES_REPLICATION_MODE
+  declare_env_alias POSTGRESQL_REPLICATION_PASSWORD POSTGRES_REPLICATION_PASSWORD
+  declare_env_alias POSTGRESQL_REPLICATION_USER POSTGRES_REPLICATION_USER
+  declare_env_alias POSTGRESQL_SYNCHRONOUS_COMMIT_MODE POSTGRES_SYNCHRONOUS_COMMIT_MODE
+  declare_env_alias POSTGRESQL_PASSWORD_FILE POSTGRES_PASSWORD_FILE
+  declare_env_alias POSTGRESQL_REPLICATION_PASSWORD_FILE POSTGRES_REPLICATION_PASSWORD_FILE
+  declare_env_alias POSTGRESQL_INIT_MAX_TIMEOUT POSTGRES_INIT_MAX_TIMEOUT
+
+    cat <<"EOF"
+# Paths
+export POSTGRESQL_VOLUMEDIR="/bitnami/postgresql"
+# POSTGRESQL_DATA_DIR used for backwards compatibility
+export POSTGRESQL_DATADIR="${POSTGRESQL_DATA_DIR:-$POSTGRESQL_VOLUMEDIR/data}"
+export POSTGRESQL_BASEDIR="/opt/bitnami/postgresql"
+export POSTGRESQL_CONFDIR="$POSTGRESQL_BASEDIR/conf"
+export POSTGRESQL_MOUNTED_CONFDIR="/bitnami/postgresql/conf"
+export POSTGRESQL_CONFFILE="$POSTGRESQL_CONFDIR/postgresql.conf"
+export POSTGRESQL_PGHBAFILE="$POSTGRESQL_CONFDIR/pg_hba.conf"
+export POSTGRESQL_RECOVERYFILE="$POSTGRESQL_DATADIR/recovery.conf"
+export POSTGRESQL_LOGDIR="$POSTGRESQL_BASEDIR/logs"
+export POSTGRESQL_LOGFILE="$POSTGRESQL_LOGDIR/postgresql.log"
+export POSTGRESQL_TMPDIR="$POSTGRESQL_BASEDIR/tmp"
+export POSTGRESQL_PIDFILE="$POSTGRESQL_TMPDIR/postgresql.pid"
+export POSTGRESQL_BINDIR="$POSTGRESQL_BASEDIR/bin"
+export POSTGRESQL_INITSCRIPTS_DIR=/docker-entrypoint-initdb.d
+export PATH="$POSTGRESQL_BINDIR:$PATH"
```

Comment:
I see we're using on other Bash containers:

- For files

```
XXX_CONF_FILE
XXX_PID_FILE
...
```

- For directories:

```
XXX_DATADIR
XXX_CONFDIR
...
```

We should be consistent on this and adapt this, or other bash containers. My vote, everything moved to `XXX_YYY_DIR` and `XXX_ZZZ_FILE`